tools/gnur-runtime/src/main/java/org/renjin/gnur/api/Rinternals.java

```diff
@@ -1960,7 +1960,7 @@ public static BytePtr Rf_translateCharUTF8(SEXP x) {
    * @return The SEXPTYPE's name within R.
    */
   public static BytePtr Rf_type2char(/*SEXPTYPE*/ int st) {
-    throw new UnimplementedGnuApiMethod("Rf_type2char");
+    return BytePtr.fromString(SexpType.typeName(st));
```

Comment:
I think a null terminated string is expected here. Use [BytePtr.nullTerminatedString(String, Charset)](http://javadoc.renjin.org/latest/org/renjin/gcc/runtime/BytePtr.html#nullTerminatedString(java.lang.String,%20java.nio.charset.Charset))